### PR TITLE
Add third-party identifiers

### LINKED ISSUE #249: [Feature Request] Store third-party identifiers
Implement storage of 3rd party IDs for tracks, artists, and albums. Common IDs:

* [Musicbrainz MBID](https://musicbrainz.org/doc/MusicBrainz_Identifier)
* [Spotify URI/ID](https://developer.spotify.com/documentation/web-api/concepts/spotify-uris-ids)

Benefits of storage:

* Makes album/artist artwork lookup require less api calls
* Can serve "external links" for spotify directly instead of using search
* Enables easier duplicate detection
* Enables easier re-parsing and metadata lookup in the future

The IDs could be stored on a new table that is a simple KV store with a FK/discriminator for the entity it is associated with. The ID should be able to be submitted alongside a new scrobble (free lookup!) since some scrobble clients can do this. Additionally, it should be able to be edited/updated through an API endpoint.

### PR DESCRIPTION
**Pull request by duckfromdiscord**
Adds functionality for third-party identifiers to maloja.

The goals: allow a scrobble client to embed identifiers (like Spotify ID, musicbrainz) with scrobble data, and allow them to be read as well.

Will fix #249.

### CODE REVIEW

**Comment by duckfromdiscord** [comment]
I am trying to decide how to store the identifiers in the database. On one hand, [the scrobble dictionary schema cannot/should not be changed](https://github.com/krateng/maloja/blob/c6cf28896ca836407a1943cae5542d2b83d009cf/maloja/database/sqldb.py#L194), so we cannot include a new `identifiers` column. On the other hand, the precedent set by #149 is that identifiers should go into the `extra` field, but the `extra` field [is hidden unless `include_internal` is specified](https://github.com/krateng/maloja/blob/c6cf28896ca836407a1943cae5542d2b83d009cf/maloja/database/sqldb.py#L259).

I think the second option makes more sense, at least for storing the data. Retrieving the data is much more difficult, because we either include all of the `include_internal` data or none of it, and there is no parameter to `/scrobbles` that allows us to do so in the first place. We also cannot return the identifiers as a new tag in the API response, because we are returning the scrobble dict directly. So this is why, as @FoxxMD said, the scrobbles endpoint does not display stored musicbrainz tags.

Also, once identifier storing/retrieval is finished, it might make sense to move the musicbrainz tags alongside them. This also brings up the issue of potentially coming up with a set of standard key "names" for each identifier. For example `identifiers['release_mbid']`, and now `identifiers['spotify_track_id']` maybe?

tl;dr: To display identifiers, no matter how we implement them, we need to include them in the scrobble dict. The question is how. We must either create a new column in the database, or add an option to pass-through the `extra` field when querying scrobbles through the API.

**Comment by duckfromdiscord** [comment]
The "standard key names" for MusicBrainz could be [Picard's internal names](https://picard-docs.musicbrainz.org/downloads/MusicBrainz_Picard_Tag_Map.html). Spotify should probably just be `spotify_track_id`, `spotify_album_id` etc.
